src/vaults/PoolManager.sol

```diff
@@ -98,10 +100,23 @@ contract PoolManager is Auth, Recoverable, IPoolManager, IUpdateContract, IPoolM
     /// @inheritdoc IPoolManager
     function transferShares(uint16 centrifugeId, uint64 poolId, bytes16 scId, bytes32 receiver, uint128 amount)
         external
-        auth
+        payable
     {
-        IShareToken shareToken_ = IShareToken(shareToken(poolId, scId));
-        shareToken_.burn(msg.sender, amount);
+        IShareToken share = IShareToken(shareToken(poolId, scId));
+        require(
+            share.checkTransferRestriction(msg.sender, address(uint160(centrifugeId)), amount),
+            CrossChainTransferNotAllowed()
+        );
+
+        gateway.payTransaction{value: msg.value}(msg.sender);
+
+        try share.authTransferFrom(msg.sender, msg.sender, address(this), amount) returns (bool) {}
```

Comment:
Matching behaviour from the async redeem vault logic

Reply:
We coudl also have removed the need for an allowance on `burn()` and used this directly, but this would have required migrating the existing deployed share tokens which is not possible.